packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -0,0 +1,210 @@
+import classNames from 'classnames';
+import * as VaporSVG from 'coveo-styleguide';
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {InputConnected, Svg} from 'react-vapor';
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {changeInputValue} from '../input/InputActions';
+import {InputSelectors} from '../input/InputSelectors';
+
+export interface LimitOwnProps {
+    id?: string;
+    title: string;
+    limitLabel?: string;
+    usage?: number;
+    limit?: number;
+    isLimitEditable?: boolean;
+    isLimitTheGoalToReach?: boolean;
+    isHistoryIncluded?: boolean;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: LimitOwnProps) => {
+    const ownLimit: string = ownProps.limit.toString() || Limit.defaultProps.limit.toString();
+    const currentLimit: string = InputSelectors.getValue(state, {id: ownProps.id}) || ownLimit;
+    return {
+        currentLimit,
+    };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: LimitOwnProps) => ({
+    onChangeLimit: (limitValue: string) => {
+        dispatch(changeInputValue(ownProps.id, limitValue, true));
+    },
+});
+
+export interface LimitProps
+    extends LimitOwnProps,
+        ReturnType<typeof mapDispatchToProps>,
+        ReturnType<typeof mapStateToProps> {}
+
+export const LimitDisconnect: React.FunctionComponent<LimitProps> = ({
+    id,
+    title,
+    limitLabel,
+    usage,
+    limit,
+    currentLimit,
+    isLimitEditable,
+    isLimitTheGoalToReach,
+    isHistoryIncluded,
+    onChangeLimit: onChangeLimit,
+}) => {
+    return (
+        <div className="limit-box mb2">
+            <div className="limit-box-main p2 pb1">
+                <HeaderDivision limitTitle={title} isHistory={isHistoryIncluded} />
+                <ContentDivision
+                    id={id}
+                    usageValue={usage}
+                    limitValue={limit}
+                    isEditable={isLimitEditable}
+                    limitTitle={limitLabel}
+                    onChangeLimit={onChangeLimit}
+                />
+            </div>
+            <ProgressBar usageValue={usage} isLimitTheGoal={isLimitTheGoalToReach} limitValue={+currentLimit} />
```

Comment:
is the `+currentLimit` a typo or something i never saw?

Reply:
It's something you never saw. Now that the limit component is connected, this variable represent the prop that mapped the state of the input. This value is mainly used to change the progress bar class and move the position of the latter based on the user's limit choice. 
As the input value is a string, I had to convert currentLimit to a number (which explains the + in front).

Reply:
today i learned ! :D